netty/src/test/java/io/grpc/transport/netty/NettyClientTransportTest.java

```diff
@@ -0,0 +1,147 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.transport.netty;
+
+import static io.netty.handler.codec.http2.Http2CodecUtil.DEFAULT_WINDOW_SIZE;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import io.grpc.Metadata;
+import io.grpc.testing.TestUtils;
+import io.grpc.transport.ClientTransport;
+import io.grpc.transport.ServerListener;
+import io.grpc.transport.ServerStream;
+import io.grpc.transport.ServerStreamListener;
+import io.grpc.transport.ServerTransport;
+import io.grpc.transport.ServerTransportListener;
+
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.ssl.SslContext;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.File;
+import java.net.InetSocketAddress;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Tsests for {@link NettyClientTransport}.
+ */
+@RunWith(JUnit4.class)
+public class NettyClientTransportTest {
+
+  @Mock
+  private ServerListener serverListener;
+
+  @Mock
+  private ServerTransportListener serverTransportListener;
+
+  @Mock
+  private ServerStreamListener serverStreamListener;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    when(serverListener.transportCreated(any(ServerTransport.class))).thenReturn(
+            serverTransportListener);
+    when(serverTransportListener.streamCreated(any(ServerStream.class), anyString(),
+            any(Metadata.Headers.class))).thenReturn(serverStreamListener);
+  }
+
+  /**
+   * Verifies that we can create multiple TLS client transports from the same builder.
+   */
+  @Test
+  public void creatingMultipleTlsTransportsShouldSucceed() throws Exception {
+    int port = TestUtils.pickUnusedPort();
+    InetSocketAddress address = TestUtils.overrideHostFromTestCa("localhost", port);
+
+    File serverCert = TestUtils.loadCert("server1.pem");
+    File key = TestUtils.loadCert("server1.key");
+    SslContext serverContext = GrpcSslContexts.forServer(serverCert, key).build();
+    NettyServer server = new NettyServer(address, NioServerSocketChannel.class,
+            new NioEventLoopGroup(),new NioEventLoopGroup(), serverContext, 100,
```

Comment:
You have to shut down the event loop groups, otherwise they continue running.

Can you specify the argument `1` to the loops so tons of threads aren't created? Also, feel free to reuse the event loop between the boss and workers.